core/src/main/scala/chisel3/RelativeOrientation.scala

```diff
@@ -0,0 +1,98 @@
+package chisel3
+
+import chisel3.experimental.DataMirror
+
+
+// Indicates whether the active side is aligned or flipped relative to the active side's root
+sealed trait RelativeOrientation { 
```

Comment:
I think functions in this trait is user api?
maybe need more scaladoc to functions.